core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2092,6 +2092,7 @@ class SparkContext(config: SparkConf) extends Logging {
    * @param exitCode Specified exit code that will passed to scheduler backend in client mode.
    */
   def stop(exitCode: Int): Unit = {
+    logInfo(s"SparkContext is stopping with exitCode $exitCode.")
     if (LiveListenerBus.withinListenerThread.value) {
       throw new SparkException(s"Cannot stop SparkContext within listener bus thread.")
     }
```

Comment:
Do we need to move it after `if (LiveListenerBus.withinListenerThread.value)`? i.e.,

```suggestion
    if (LiveListenerBus.withinListenerThread.value) {
      throw new SparkException(s"Cannot stop SparkContext within listener bus thread.")
    }
    logInfo(s"SparkContext is stopping with exitCode $exitCode.")
```

Reply:
Thank you for reviewing. I also consider that and this ways, but I want to be clear that this log is a marker of this method itself. So, it should be at the first line of the method before Spark doing anything.

Reply:
For example, the exception `SparkException(s"Cannot stop SparkContext within listener bus thread.")` should be thrown after this log, @viirya .

Reply:
Okay, makes sense.